graph/db/graph.go

```diff
@@ -1178,6 +1178,80 @@ func (c *ChannelGraph) addChannelEdge(tx kvdb.RwTx,
 	return chanIndex.Put(b.Bytes(), chanKey[:])
 }
 
+// ReAddChannelEdge removes the edge with the given channel ID from the
+// database and adds the new edge to guarantee atomicity.
+// This is important for option-scid-alias channels which may change its SCID
+// over the course of its lifetime (e.g., public zero-conf channels).
+func (c *ChannelGraph) ReAddChannelEdge(
```

Comment:
A typical flow for adding an edge,
1. `ChannelAnnouncement` msg received by the gossiper
2. the gossiper validates it and sends it to the channel graph
3. the channel graph validates and adds it to the graph db

And a similar flow is used for `ChannelUpdate`, but now we just bypass all the validations here?

Reply:
I think it is ok to skip the validation for the `ChannelAnnouncement` we created ourselves ? Why should we not trust our own ChannelAnnouncment and ChannelUpdate when its internally sourced ?